src/General/Index.coffee

```diff
@@ -744,25 +764,32 @@ Index =
   sort: ->
     {liveThreadIDs, liveThreadData} = Index
     return unless liveThreadData
-    Index.sortedThreadIDs = switch Index.currentSort
-      when 'lastreply'
-        [liveThreadData...].sort((a, b) ->
-          a = num[num.length - 1] if (num = a.last_replies)
-          b = num[num.length - 1] if (num = b.last_replies)
-          b.no - a.no
-        ).map (post) -> post.no
-      when 'lastlong'
-        lastlong = (thread) ->
-          for r, i in (thread.last_replies or []) by -1
-            return r if r.com and Build.parseComment(r.com).replace(/[^a-z]/ig, '').length >= 100
-          thread
-        [liveThreadData...].sort((a, b) ->
-          lastlong(b).no - lastlong(a).no
-        ).map (post) -> post.no
-      when 'bump'       then liveThreadIDs
-      when 'birth'      then [liveThreadIDs... ].sort (a, b) -> b - a
-      when 'replycount' then [liveThreadData...].sort((a, b) -> b.replies - a.replies).map (post) -> post.no
-      when 'filecount'  then [liveThreadData...].sort((a, b) -> b.images  - a.images ).map (post) -> post.no
+    Index.sortedThreadIDs = do ->
+      if Index.currentSort.slice(-4) == "-rev"
+        currentSort = Index.currentSort.slice(0,-4)
+        reverse = true
+      else
+        currentSort = Index.currentSort
+      sorted = switch currentSort
+        when 'lastreply'
+          [liveThreadData...].sort((a, b) ->
+            a = num[num.length - 1] if (num = a.last_replies)
+            b = num[num.length - 1] if (num = b.last_replies)
+            b.no - a.no
+          ).map (post) -> post.no
+        when 'lastlong'
+          lastlong = (thread) ->
+            for r, i in (thread.last_replies or []) by -1
+              return r if r.com and Build.parseComment(r.com).replace(/[^a-z]/ig, '').length >= 100
+            thread
+          [liveThreadData...].sort((a, b) ->
+            lastlong(b).no - lastlong(a).no
+          ).map (post) -> post.no
+        when 'bump'       then liveThreadIDs
+        when 'birth'      then [liveThreadIDs... ].sort (a, b) -> b - a
+        when 'replycount' then [liveThreadData...].sort((a, b) -> b.replies - a.replies).map (post) -> post.no
+        when 'filecount'  then [liveThreadData...].sort((a, b) -> b.images  - a.images ).map (post) -> post.no
+      return if reverse then [sorted...].reverse() else sorted
```

Comment:
The `do ->` block isn't needed here, as far as I can tell; it would be simpler to just assign what you're returning to `Index.sortedThreadIDs` at the end.